interop-tests/src/test/scala/akka/grpc/scaladsl/PowerApiSpec.scala

```diff
@@ -84,5 +109,41 @@ class PowerApiSpec(backend: String)
       response.headers.getText("baz").get should be("qux")
       response.trailers.futureValue.getText("foo").get should be("bar")
     }
+
+    "(on streamed calls) redeem the headers future as soon as they're available (and trailers future when trailers arrive)" in {
+      // invoking streamed calls using the power API materializes a Future[GrpcResponseMetadata]
+      // that should redeem as soon as the HEADERS is consumed. Then, the GrpcResponseMetadata instance
+      // contains another Future that will redeem when receiving the trailers.
+      client = GreeterServiceClient(
+        GrpcClientSettings.connectToServiceAt("localhost", server.localAddress.getPort).withTls(false))
+
+      val request = HelloRequest("Alice")
+      val responseSource: Source[HelloReply, Future[GrpcResponseMetadata]] =
+        client.itKeepsReplying().invokeWithMetadata(request)
```

Comment:
`streamHellos` might have been even easier to be able to 'control' the timing of the responses (and response completion) from the test, but this is :+1: as well

Reply:
I completely missed this comment yesterday. I'll keep it mind.